llvm/test/CodeGen/RISCV/cheri/compress.ll

```diff
@@ -0,0 +1,118 @@
+; This test is designed to run twice, once with function attributes and once
+; with target attributes added on the command line.
+; See compress.ll in the folder above
+;
+; RUN: cat %s > %t.tgtattr
+; RUN: echo 'attributes #0 = { nounwind }' >> %t.tgtattr
+; RUN: %riscv32_cheri_purecap_llc -mattr=+c,+xcheri,+cap-mode -filetype=obj < %t.tgtattr \
+; RUN:   | llvm-objdump -d -M no-aliases --syms - | FileCheck %s
+; RUN: %riscv64_cheri_purecap_llc -mattr=+c,+xcheri,+cap-mode -filetype=obj < %t.tgtattr \
+; RUN:   | llvm-objdump -d -M no-aliases --syms - | FileCheck %s
+; RUN: %riscv64_cheri_purecap_llc -mattr=+c,+xcheri,+cap-mode,+xcheri-norvc -filetype=obj < %t.tgtattr \
+; RUN:   | llvm-objdump -d -M no-aliases --syms - | FileCheck %s --check-prefix=NORVC
+
+; RUN: cat %s > %t.fnattr
+; RUN: echo 'attributes #0 = { nounwind "target-features"="+c,+xcheri,+cap-mode" }' >> %t.fnattr
+; RUN: %riscv32_cheri_purecap_llc -filetype=obj < %t.fnattr \
+; RUN:   | llvm-objdump -d --mattr=+c -M no-aliases --syms - | FileCheck %s
+; RUN: %riscv64_cheri_purecap_llc -filetype=obj < %t.fnattr \
+; RUN:   | llvm-objdump -d --mattr=+c -M no-aliases --syms - | FileCheck %s
+; RUN: cat %s > %t.fnattr
+; RUN: echo 'attributes #0 = { nounwind "target-features"="+c,+xcheri,+cap-mode,+xcheri-norvc" }' >> %t.fnattr
+; RUN: %riscv64_cheri_purecap_llc -filetype=obj < %t.fnattr \
+; RUN:   | llvm-objdump -d --mattr=+c -M no-aliases --syms - | FileCheck %s  --check-prefix=NORVC
+
+; Basic check that we can use CHERI compressed instructions
+
+; Check the size of the functions to check whether we used compressed instructions:
```

Comment:
Why? That's implied by the disassembly...

Reply:
I was concerned that checking for `clc ...` would also match `c.clc` and I couldn't figure out how to get FileCheck to match a tab before the mnemonic.